src/system.h

```diff
@@ -16,21 +17,21 @@ struct Image {
 
     long last_display = 0;
 
-    Image(wxBitmap _bm, uint64_t _hash, double _sc, vector<uint8_t> &&pd)
-        : bm_orig(_bm), png_data(std::move(pd)), hash(_hash), display_scale(_sc) {}
+    Image(wxBitmap _bm, uint64_t _hash, double _sc, vector<uint8_t> &&pd, const wxBitmapType &imt)
+        : bm_orig(_bm), hash(_hash), display_scale(_sc), image_data(std::move(pd)), image_type(imt) {}

```

Comment:
Image class gets expanded by wxBitmapType image_type and renaming of png_data to image_data (more generic).